app/models/ext_management_system.rb

```diff
@@ -426,6 +426,34 @@ def enable!
     update!(:enabled => true)
   end
 
+  # override destroy_queue from AsyncDeleteMixin
+  def destroy_queue(deliver_on = Time.now)
+    _log.info("Queuing destroy of #{self.class.name} with id: #{id}")
+    MiqQueue.put(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "orchestrate_destroy",
+      :deliver_on  => deliver_on
+    )
+  end
+
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    if self.destroy == nil
+      destroy_queue(15.seconds.from_now)
```

Comment:
I tried with `raise MiqException::MiqQueueRetryLater.new(:deliver_on => 15.seconds.from_now)`
but it would just queue the job immediatly. It did write a fancy message from https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_queue.rb#L328 though.
I cant find the correct way to utilize that.